File: openclaw-android/app/src/main/java/ai/openclaw/android/domain/model/Session.kt
package ai.openclaw.android.domain.model

/**
 * 会话模型
 */
data class Session(
    val key: String,
    val label: String?,
    val channel: String?,
    val provider: String?,
    val model: String?,
    val createdAt: Long,
    val updatedAt: Long,
    val messageCount: Int
) {
    val displayName: String
        get() = label ?: "Session ${key.take(8)}"
}

File: lib/src/core/models/session.dart
/// Session model
library;

import 'package:freezed_annotation/freezed_annotation.dart';

part 'session.freezed.dart';
part 'session.g.dart';

/// Chat session
@freezed
class Session with _$Session {
  const factory Session({
    required String key,
    String? label,
    String? agentId,
    String? lastMessage,
    DateTime? createdAt,
    DateTime? updatedAt,
    DateTime? lastActiveAt,
    @Default(false) bool isArchived,
    @Default(false) bool isPinned,
    @Default(0) int messageCount,
  }) = _Session;

  factory Session.fromJson(Map<String, dynamic> json) =>
      _$SessionFromJson(json);
}